docs/react/guides/migrating-to-v5.md

```diff
@@ -341,6 +341,10 @@ useInfiniteQuery({
 
 Previously, we've allowed to overwrite the `pageParams` that would be returned from `getNextPageParam` or `getPreviousPageParam` by passing a `pageParam` value directly to `fetchNextPage` or `fetchPreviousPage`. This feature didn't work at all with refetches and wasn't widely known or used. This also means that `getNextPageParam` is now required for infinite queries.
 
+### Returning `null` from `getNextPageParam` or `getPreviousPageParam` now indicates that there is no further page available
+
+In v4, you needed to explicitly return `undefined` to indicate that there is no further page available. We've widened this check to include `null`.
```

Comment:
technically not though ;)

https://github.com/TanStack/query/blob/100ae76c402f862f4702eac30369340afd5b57c1/packages/query-core/src/infiniteQueryBehavior.ts#L194

Reply:
yeah but only for `hasNextPage` and `hasPreviousPage` checks. The actual fetching happened here:

https://github.com/TanStack/query/blob/100ae76c402f862f4702eac30369340afd5b57c1/packages/query-core/src/infiniteQueryBehavior.ts#L76

and there, we only checked for `undefined` :) It was pretty inconsistent and buggy. If you returned `null`, it would:
- tell you that there is no next page
- but it would still continue to fetch if you called `fetchNextPage` 🤷 

btw, I adapted that code path, too to include null now.

Reply:
oh damn 😅